core/data-transformers/product-options-transformer.ts

```diff
@@ -0,0 +1,185 @@
+              type: 'radio-group',
+              label: option.displayName,
+              required: option.isRequired,
+              name: option.entityId.toString(),
+              defaultValue: values.find((value) => value.isDefault)?.entityId.toString(),
+              options: values.map((value) => ({
+                label: value.label,
+                value: value.entityId.toString(),
+              })),
+            };
+          }
+
+          case 'DropdownList': {
+            return {
+              persist: option.isVariantOption,
+              type: 'select',
+              label: option.displayName,
+              required: option.isRequired,
+              name: option.entityId.toString(),
+              defaultValue: values.find((value) => value.isDefault)?.entityId.toString(),
+              options: values.map((value) => ({
+                label: value.label,
+                value: value.entityId.toString(),
+              })),
+            };
+          }
+
+          case 'ProductPickList':
+          case 'ProductPickListWithImages': {
+            return {
+              persist: option.isVariantOption,
+              type: 'card-radio-group',
+              label: option.displayName,
+              required: option.isRequired,
+              name: option.entityId.toString(),
+              defaultValue: values.find((value) => value.isDefault)?.entityId.toString(),
+              options: values
+                .filter(
+                  (value) =>
+                    '__typename' in value && value.__typename === 'ProductPickListOptionValue',
+                )
+                .map((value) => ({
+                  label: value.label,
+                  value: value.entityId.toString(),
+                  image: {
+                    src: value.defaultImage?.url ?? '',
+                    alt: value.defaultImage?.altText ?? '',
+                  },
+                })),
+            };
+          }
+
+          default:
+            return null;
+        }
+      }
+
+      if (option.__typename === 'CheckboxOption') {
+        return {
+          persist: option.isVariantOption,
+          type: 'checkbox',
+          label: option.displayName,
+          required: option.isRequired,
+          name: option.entityId.toString(),
+          defaultValue: option.checkedByDefault.toString(),
+          uncheckedValue: option.uncheckedOptionValueEntityId.toString(),
+          checkedValue: option.checkedOptionValueEntityId.toString(),
+        };
+      }
+
+      if (option.__typename === 'NumberFieldOption') {
+        return {
+          persist: option.isVariantOption,
+          type: 'number',
+          label: option.displayName,
+          required: option.isRequired,
+          name: option.entityId.toString(),
+          defaultValue: option.defaultNumber?.toString(),
+          min: option.lowest ?? undefined,
+          max: option.highest ?? undefined,
+          // TODO: other props?
```

Comment:
🍹 Can we add a bit more detail here? Not sure what "other props?" means. Is it about extending the VIBES field to support Catalyst-only props using generics?